### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by fernandosfreitas** [dynamicslab/pysindy#348]
Hello,

I'm starting to study SINDy and I'm following the examples that were published by the group. I made an attempt to include the time (variable t) in the custom library. In the **output** of the code the variable t is appearing 3 times. Why ?

### Reproducing code example:

...
t, x, y, z = gen_data1()
X = np.stack((x, y, z), axis=-1)  # First column is x, second is y
differentiation_method = ps.FiniteDifference(order=2)
poly_lib = ps.PolynomialLibrary(degree=2)
**library_functions = [lambda t: t]
library_function_names = [lambda t: 't']**
custom_lib = ps.CustomLibrary( library_functions=library_functions, function_names=library_function_names )
generalized_lib = ps.GeneralizedLibrary( [poly_lib, custom_lib])
feature_library = generalized_lib
optimizer = ps.STLSQ(threshold=0.2)
model = ps.SINDy(
    differentiation_method=differentiation_method,
    feature_library=feature_library,
    optimizer=optimizer,
    feature_names=["x", "y", "z", "t"],)
model.fit(X, t=t)
model.print()
print(model.get_feature_names())
```

### **Output:**

(x)' = 0.441 1 + 2.346 x + 2.346 t
(y)' = -0.684 x + 0.897 y + -0.897 z + -0.684 t + 0.897 t + -0.897 t
(z)' = 0.684 x + -0.897 y + 0.897 z + 0.684 t + -0.897 t + 0.897 t
['1', 'x', 'y', 'z', 'x^2', 'x y', 'x z', 'y^2', 'y z', 'z^2', 't', 't', 't']

### PySINDy/Python version information:

1.7.5 3.8.6 (tags/v3.8.6:db45529, Sep 23 2020, 15:52:53) [MSC v.1927 64 bit (AMD64)]

**Comment by Jacob-Stevens-Haas** [comment]
In part because your linear term is in both the CustomLibrary and the Polynomial Library.  However, that code doesn't look like it would produce that output, even with the stuff that's missing.  Post a complete minimal working example.

**Comment by znicolaou** [comment]
Ah, I see the issue here. 

It is because your input data `X` has three features `x`, `y`, and `z`. PySINDy will inspect the number of arguments to the library functions, then produce terms with each possible combination of the input features as the arguments. So each of `x`, `y`, and `z` is being passed through the library function `lambda t: t` as the local input `t`. What you actually want to do is use a library function with no inputs:
```
library_functions = [lambda: t]
library_function_names = [lambda: 't']
```
This should resolve the variable `t` in the global scope rather than as a local argument.

**Comment by fernandosfreitas** [comment]
Thank you for the explanation. 

I can understand that the 3 coefficients of the variable t must be added. 
Like this:

### **Output:**

(x)' = 0.441 1 + 2.346 x + 2.346 t
(y)' = -0.684 x + 0.897 y + -0.897 z + -0.684 t + **0.897 t + -0.897 t**
(z)' = 0.684 x + -0.897 y + 0.897 z + 0.684 t + **-0.897 t + 0.897 t**

It's actually:

(x)' = 0.441 1 + 2.346 x + 2.346 t
(y)' = -0.684 x + 0.897 y + -0.897 z + -0.684 t    (because the terms in bold cancel each other out)
(z)' = 0.684 x + -0.897 y + 0.897 z + 0.684 t
